server/types.ts

```diff
@@ -0,0 +1,153 @@
+/*
```

Comment:
What's general best practice - should `types.ts` be per folder/barrel file? Or one at top level?

Reply:
I havent seen anything advocating one approach vs another. These types/interfaces are used across the server code. If there was a one time typed function, I added the definition ahead/as a part of the function. My starter for 10 was types per area (ie client/server), and one at the repo/util level if needed for both.